### PR TITLE
Update DG

### PR DESCRIPTION
**Pull request by WinSheng1**

### CODE REVIEW

File: docs/diagrams/EditApplicantSequenceDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+end box
+
+box Model MODEL_COLOR_T1
+participant ":Model" as Model MODEL_COLOR
+end box
+
+[-> LogicManager : execute("edita 1 n/Johnny Doe p/91234567")
+activate LogicManager
+
+LogicManager -> AddressBookParser : parseCommand("edita 1 n/Johnny Doe p/91234567")
```
Comment (Saezenn) [inline docs/diagrams/EditApplicantSequenceDiagram.puml]: should be name/ and phone/ ?

File: docs/diagrams/EditApplicantSequenceDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+participant ":EditApplicantCommandParser" as EditApplicantCommandParser LOGIC_COLOR
+participant "d:EditApplicantCommand" as EditApplicantCommand LOGIC_COLOR
+participant ":CommandResult" as CommandResult LOGIC_COLOR
+end box
+
+box Model MODEL_COLOR_T1
+participant ":Model" as Model MODEL_COLOR
+end box
+
+[-> LogicManager : execute("edita 1 n/Johnny Doe p/91234567")
```
Comment (Saezenn) [inline docs/diagrams/EditApplicantSequenceDiagram.puml]: here as well, /name and /phone?

File: docs/diagrams/EditApplicantSequenceDiagram.puml (hunk 1 above)
Comment (WinSheng1) [reply docs/diagrams/EditApplicantSequenceDiagram.puml]: Oh right, thank you

File: docs/images/EditApplicantSequenceDiagram.png
Comment (jonasongg) [inline docs/images/EditApplicantSequenceDiagram.png:1]: might want to rename the EditApplicantCommand object from d to e, then change the 3 return objects to e as well!

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -263,35 +263,25 @@ The sequence diagram below also shows the interaction between the various compon
...
-  details provided, effectively modifying the existing applicant's information.
-* Step 5: After the execution of the `EditApplicantCommand`, the applicant's details are successfully edited in the
-  applicant list.
+### Edit a `Member`/`Applicant`
+
+Edits the details of an existing `Member`/ `Applicant` identified by their `MEMBER_INDEX`/`APPLICANT_INDEX` in the displayed member/applicant list.
+The commands are implemented in the `EditMemberCommand` and `EditApplicantCommand` classes which extend the `Command` class.
+
+* Step 1. The `EditMemberCommand`/`EditApplicantCommand` object's `execute()` method is called.
+* Step 2. The `MEMBER_INDEX`/`APPLICANT_INDEX` is checked to be within the valid range of the displayed member/applicant list. If the `MEMBER_INDEX`/`APPLICANT_INDEX` given is invalid(i.e out of range), a `CommandException` is thrown.
```
Comment (AlfredBeNoel) [inline docs/DeveloperGuide.md:272]: Might be good to change to say `Member` or `Applicant` list

**Review by AlfredBeNoel — COMMENTED** [review]
LGTM.

**Pull request merged**